go/cmd/dolt/commands/diff.go

```diff
@@ -560,36 +536,146 @@ func diffRows(ctx context.Context, newRows, oldRows types.Map, newSch, oldSch sc
 		return errhand.BuildDError("Error diffing: %v", err.Error()).Build()
 	}
 
-	return verr
+	if badRowVErr != nil {
+		return badRowVErr
+	}
+
+	return nil
+}
+
+func buildPipeline(dArgs *diffArgs, joiner *rowconv.Joiner, ds *diff.DiffSplitter, untypedUnionSch schema.Schema, src *diff.RowDiffSource, sink *diff.ColorDiffSink, badRowCB pipeline.BadRowCallback) (*pipeline.Pipeline, errhand.VerboseError) {
+	var where FilterFn
+	var selTrans *SelectTransform
+	where, err := ParseWhere(joiner.GetSchema(), dArgs.where)
+
+	if err != nil {
+		return nil, errhand.BuildDError("error: failed to parse where cause").AddCause(err).SetPrintUsage().Build()
+	}
+
+	transforms := pipeline.NewTransformCollection()
```

Comment:
Might be more readable to use construct the pipeline using pipeline.NewPartialPipeline, then call AddStage and SetOutput on it as you go